lib/trie/trie_test.go

```diff
@@ -91,101 +91,6 @@ func Test_Trie_Snapshot(t *testing.T) {
 	assert.Equal(t, expectedTrie, newTrie)
 }
 
-func Test_Trie_maybeUpdateGeneration(t *testing.T) {
```

Comment:
do we want to still have a test for `updateGeneration`?

Reply:
Re-added a test for `updateGeneration`, good catch 👍 !